app/models/ability.rb

```diff
@@ -29,6 +29,11 @@ def self.define(&block)
     @@rules << block
   end
 
+  def can?(action, subject, *extra_args)
+    model = subject.try(:model) || subject
```

Comment:
I think it actually needs to be in a loop until it's no longer a decorated object, no?

Reply:
@guicassolato I do not think so, because 'model' should really mean 'model' 🤔 look at this:
![image](https://user-images.githubusercontent.com/11318903/89311058-caca6180-d675-11ea-800b-e3176a81f8c8.png)
![image](https://user-images.githubusercontent.com/11318903/89311082-d1f16f80-d675-11ea-901b-69431e8f506d.png)
![image](https://user-images.githubusercontent.com/11318903/89311098-d87fe700-d675-11ea-811f-f1098e987fbc.png)
![image](https://user-images.githubusercontent.com/11318903/89311114-dddd3180-d675-11ea-9587-311c6686b7b2.png)

Reply:
https://github.com/GoodMeasuresLLC/draper-cancancan/pull/1

Reply:
It's ironic because apparently it is currently working in that scenario (it is a collection in this case). The question is 'why' xD
I will check it to ensure that it is stable, and I may add a test.

Reply:
@guicassolato you are right about this bug.

The reason why 'it is apparently working although it is a collection' is that it is checking the `can?` for each entry of the collection. 

The solution tho is not the loop of `subject.model` 🤔 that's not working 🤔 
I am on it 👍

Reply:
Fixed and extracted to its own PR https://github.com/3scale/porta/pull/2103